swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -83,227 +70,145 @@ class Adapter {
     self.token = token
     self.packetTunnelProvider = packetTunnelProvider
     self.callbackHandler = CallbackHandler(logger: packetTunnelProvider.logger)
-    self.state = .stoppedTunnel
+    self.state = .tunnelStopped
     self.logFilter = logFilter
     self.connlibLogFolderPath = SharedAccess.connlibLogFolderURL?.path ?? ""
-    self.firezoneIdFileURL = SharedAccess.baseFolderURL.appendingPathComponent("firezone-id")
     self.logger = packetTunnelProvider.logger
+    self.networkSettings = nil
   }
 
+  // Could happen abruptly if the process is killed.
   deinit {
-    self.logger.log("Adapter.deinit")
+    logger.log("Adapter.deinit")
+
     // Cancel network monitor
     networkMonitor?.cancel()
 
     // Shutdown the tunnel
-    if case .tunnelReady(let session) = self.state {
+    if case .tunnelStarted(let session) = self.state {
       logger.log("Adapter.deinit: Shutting down connlib")
       session.disconnect()
     }
   }
 
-  /// Start the tunnel tunnel.
+  /// Start the tunnel.
   /// - Parameters:
   ///   - completionHandler: completion handler.
   public func start(completionHandler: @escaping (AdapterError?) -> Void) throws {
-    workQueue.async { [weak self] in
-      guard let self = self else { return }
-
-      self.logger.log("Adapter.start")
-      guard case .stoppedTunnel = self.state else {
-        logger.error("\(#function): Invalid Adapter state")
-        completionHandler(.invalidState)
-        return
-      }
+    logger.log("Adapter.start")
+    guard case .tunnelStopped = self.state else {
+      logger.error("\(#function): Invalid Adapter state")
+      completionHandler(.invalidState)
+      return
+    }
 
-      self.callbackHandler.delegate = self
+    callbackHandler.delegate = self
 
-      if self.connlibLogFolderPath.isEmpty {
-        self.logger.error("Cannot get shared log folder for connlib")
-      }
+    if connlibLogFolderPath.isEmpty {
+      logger.error("Cannot get shared log folder for connlib")
+    }
 
-      self.logger.log("Adapter.start: Starting connlib")
-      do {
-        // We can only get the system's default resolvers before connlib starts, and then they'll
-        // be overwritten by the ones from connlib. So cache them here for getSystemDefaultResolvers
-        // to retrieve them later.
-        self.callbackHandler.setSystemDefaultResolvers(
-          resolvers: Resolv().getservers().map(Resolv.getnameinfo)
-        )
-        self.state = .startingTunnel(
-          session: try WrappedSession.connect(
-            self.controlPlaneURLString,
-            self.token,
-            self.getOrCreateFirezoneId(from: self.firezoneIdFileURL),
-            self.getDeviceName(),
-            self.getOSVersion(),
-            self.connlibLogFolderPath,
-            self.logFilter,
-            self.callbackHandler
-          ),
-          onStarted: completionHandler
+    self.logger.log("Adapter.start: Starting connlib")
+    do {
+      // Grab a session pointer
```

Comment:
```suggestion
      // Create a session
```

We're not getting a handle to an existing Session, right?